\documentclass{article}
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{hyperref}
\usepackage{url}
\usepackage{booktabs}
\usepackage{graphicx}
\usepackage[backend=bibtex]{biblatex}
\usepackage[margin=2cm]{geometry}
\usepackage{caption}
\usepackage{subcaption}


\def\referencepath{../../../References/references.bib}
\def\sectiontitle#1{\subsection*{\cite{#1}\citetitle{#1}\citeyear{#1}}}
\def\dc#1{$#1\,^{\circ}\mathrm{C}$}
\def\mycite#1{\citeauthor{#1} (\citeyear{#1})}

\addbibresource{\referencepath}

\title{Lab Report 1}
\author{Zachary Foster}
\date{\today}

<<setup, include=FALSE>>=
#R library imports
library(knitr)
library(xtable)
# set global chunk options
opts_chunk$set(fig.align='center', 
               fig.show='asis', 
               fig.lp="", 
               error=FALSE, 
               cach=FALSE, 
               dpi=600, 
               include=FALSE, 
               echo=FALSE, 
               tidy=TRUE, 
               out.width=".9\\linewidth", 
               fig.width = 8, 
               external=FALSE, 
               results='markup',
               message=FALSE)
#read_chunk('insert file name here for code externalization ')
@

\begin{document}

\maketitle





Dr. Muller suppiled us with contigs generated from a metagemoic assembly of DNA from a sample of marine water.
The contigs are thought to be an undescribed lineage wihtin \textit{verrucomicrobia}.
We used BLAST, GLIMMER, and web-based tools to idenitify ORFs within one of the contigs and pridict possible functions.

Using an interpolated markov model supplied by the instructors, we used GLIMMER to pridict ORFs. 
We used a custom \texttt{perl} script \texttt{parse_Glimmer.pl} to extract the amino acid sequence of the predicted ORFs.
We then made a BLAST database, using sequences supplied by the instructors, and used the database to identify the ORFs. 4 ORFs were found, 2 of which return significant BLAST hits using an -e threshold of 0.01. 
Each of the 2 ORFs with siginificant hits had one identical hit and two very weak hits. 
We then used a custom \texttt{perl} script \texttt{grab_seq.pl} to exctract one of the ORFs with and identical hit to the reference database. 
The extracted sequence was search aginst the genbank database using blastp with default settings.
All of the significant alignments the resulted that had a functional annotaion were differnt types of sulfatases. 







\begin{figure}[ht]
\centering
\begin{minipage}{.45\textwidth}
  \centering
  \includegraphics[width=.7\linewidth]{dotplot}
  \captionof{figure}{Dot plot of contigs vs. the sequence they were derived from. }
  \label{fig:dotplot}
\end{minipage}%
\hspace{10pt}%
\begin{minipage}{.45\textwidth}
  \centering
  \includegraphics[width=.7\linewidth]{coverage_hist}
  \captionof{figure}{Distribution of contig average coverage.}
  \label{fig:histogram}
\end{minipage}%
\end{figure}

\printbibliography

\end{document}
