\documentclass[12pt]{article}
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{hyperref}
\usepackage{url}
\usepackage{booktabs}
\usepackage{graphicx}
\usepackage[backend=bibtex]{biblatex}
\usepackage[margin=3cm]{geometry}
\usepackage{caption}
\usepackage{subcaption}


\def\referencepath{../../../References/references.bib}
\def\sectiontitle#1{\subsection*{\cite{#1}\citetitle{#1}\citeyear{#1}}}
\def\dc#1{$#1\,^{\circ}\mathrm{C}$}
\def\mycite#1{\citeauthor{#1} (\citeyear{#1})}

\addbibresource{\referencepath}

\title{Lab Report 2}
\author{Zachary Foster}
\date{\today}

<<setup, include=FALSE>>=
#R library imports
library(knitr)
library(xtable)
# set global chunk options
opts_chunk$set(fig.align='center', 
               fig.show='asis', 
               fig.lp="", 
               error=FALSE, 
               cach=FALSE, 
               dpi=600, 
               include=FALSE, 
               echo=FALSE, 
               tidy=TRUE, 
               out.width=".9\\linewidth", 
               fig.width = 8, 
               external=FALSE, 
               results='markup',
               message=FALSE)
#read_chunk('insert file name here for code externalization ')
@

\begin{document}

\maketitle


Dr. Muller supplied us with contigs generated from a metagenomic assembly of DNA from a sample of marine water.
The contigs are thought to be an undescribed lineage within \textit{verrucomicrobia}.
We used BLAST, GLIMMER, and web-based tools to identify ORFs within one of the contigs and predict possible functions.
Using an interpolated Markov model supplied by the instructors, we used GLIMMER to predict ORFs. 
We used a custom \texttt{perl} script \texttt{parse\_Glimmer.pl} to extract the amino acid sequence of the predicted ORFs.
We then made a BLAST database, using sequences supplied by the instructors, and used the database to identify the ORFs. 4 ORFs were found, 2 of which return significant BLAST hits using an -e threshold of 0.01. 
Each of the 2 ORFs with significant hits had one identical hit and two very weak hits. 
We then used a custom \texttt{perl} script \texttt{grab\_seq.pl} to extract one of the ORFs with and identical hit to the reference database.



The extracted sequence was search against the genbank nr database using blastp with default settings \ref{fig:blastp}.
All of the hits were hypothetical proteins and only one was full length.  
Next, a PSI blast search with 3 iterations, using default parameters, was used in order to discover more distantly related putative homologs.
This revealed many more full length matches, but all were still hypothetical proteins \ref{fig:psi-blast}.
We used the COG and CCD databases to search for conserved domains using default parameters. 
The COG database search returned a hit to a chromosome segregation ATPase \ref{fig:cog}, but the CCD databases search did not return any hits with known function \ref{fig:ccd}
Finally, we searched pfam in order to detect any protein families that the unknown sequence may be a part of, but this to return no significant result with known function. The 2 hits that were found were of unknown function \ref{fig:pfam}. From all of the analyses overall, it appears that this ORF is ether a novel protein that has not been assigned a function yet, or a conserved sequence susceptible to misassignment by automatic \textit{ab inito} gene annotating program, since there are many hits to multiple databases, but little functional information. The one significant hit with functional information was a chromosome segregation ATPase. but this alone is not sufficient evidence to predict the function of this unknown sequence. 

\begin{figure}[ht]
  \centering
  \includegraphics[width=\linewidth]{blastp}
  \captionof{figure}{Screenshot of blastp results}
  \label{fig:blastp}
\end{figure}
\begin{figure}[ht]
  \includegraphics[width=\linewidth]{psi_blast}
  \captionof{figure}{Screenshot of PSI-BLAST results}
  \label{fig:psi-blast}
\end{figure}

\begin{figure}[ht]
  \includegraphics[width=\linewidth]{cog}
  \captionof{figure}{Screenshot of COG search results.}
  \label{fig:cog}
\end{figure}
\begin{figure}[ht]
  \includegraphics[width=\linewidth]{ccd}
  \captionof{figure}{Screenshot of CCD search results}
  \label{fig:ccd}
\end{figure}
\begin{figure}[ht]
  \includegraphics[width=\linewidth]{pfam}
  \captionof{figure}{Screenshot of pfam results}
  \label{fig:pfam}
\end{figure}



\printbibliography

\end{document}
